src/models/user.js

```diff
@@ -0,0 +1,39 @@
+var bcrypt = require('bcrypt');
+
+const user = (sequelize, DataTypes) => {
+  const User = sequelize.define('user', {
+    username: {
+      type: DataTypes.STRING,
+      unique: true,
+      allowNull: false,
+      validate: {
+        notEmpty: true,
+      },
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      validate: {
+        notEmpty: true,
+        len: [7, 42],
+      },
+    },
+  });
+
+  User.associate = models => {
+    User.hasMany(models.Cart, { onDelete: 'CASCADE' });
+  };
+
+  User.beforeCreate(async user => {
+    user.password = await user.generatePasswordHash();
+  });
```

Comment:
I really appreciate you're hashing user password, bcrypt is indeed a really solid tool for the job but one thing I'm wondering here is: Does the challenge require authentication?

Also, one thing to keep in mind is that hashing on the beforeCreate hook is fine, it'll work and the database will get the hash though the password will be unencrypted throughout your request lifecycle which is bad cause you may end up printing the real user password on your logs by accident and that would be a security flaw.
Your `generatePasswordHash` looks fine to me but it should probably be called as the first thing you do on any route where the user provides credentials, this way you can do your logging just fine without worrying about spitting sensitive information by accident.

Reply:
I plan to make a simple login and logout interface and system. And the registration will be in the moment the user pays for the cart items.